presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/aggregation/GeometryStateSerializer.java

```diff
@@ -44,6 +44,8 @@ public void serialize(GeometryState state, BlockBuilder out)
     @Override
     public void deserialize(Block block, int index, GeometryState state)
     {
+        state.addMemoryUsage(-state.estimateGeometryMemorySize());
```

Comment:
Why don't we just set the memory usage with the `setGeometry` call? It's easy to miss setting the memory usage after setting the geometry.

Reply:
That's what we were doing earlier, see linked issue for details why it's problematic.